base.go

```diff
@@ -36,6 +37,14 @@ func resolve(ctx context.Context, r resolver, name string, options opts.ResolveO
 		}
 	}
 
+	if err == ErrResolveFailed {
+		i := len(name) - 1
+		for i >= 0 && name[i] != '/' {
+			i--
+		}
+		// Wrap error so that it can be tested if it is a ErrResolveFailed
+		err = fmt.Errorf("%w: %s is missing DNSLink record (https://docs.ipfs.io/concepts/dnslink/)", ErrResolveFailed, name[i+1:])
```

Comment:
As this may end up on Gateways, where we should avoid printing user-provided unsanitized input, maybe use `%q`?  It will also make it easier to spot the name.
(also, https://github.com/ipfs/go-ipfs/issues/8000#issuecomment-804230839)

```suggestion
		err = fmt.Errorf("%w: %q is missing a DNSLink record (https://docs.ipfs.io/concepts/dnslink/)", ErrResolveFailed, name[i+1:])
```